src/scan.js

```diff
@@ -188,9 +202,10 @@ function scan({
         return astray.SKIP;
       }
 
-      const componentPath = [actualFirstPart, ...restParts].join(
-        ".components."
-      );
+      const componentParts = [actualFirstPart, ...restParts];
+
+      const componentPath = componentParts.join(".components.");
+      const componentName = componentParts.join(".");
```

Comment:
This seems like it should be equivalent to the `fullCompnentName` you linked in the comment.  The `.components.` part is needed just because of the shape of the data structure, so if we strip out the `.components.` part and just join the `componentParts` with a `.`, this should reconstruct the actual value in JSX